sdk_test.go

```diff
@@ -358,6 +359,96 @@ func TestMetricsServer_WithDefaultRegistryAndDefaultPort(t *testing.T) {
 	})
 }
 
+// TestMetricsServer_WithCustomMetricsServerOpts verifies that metrics server uses custom metrics server opts.
+func TestMetricsServer_WithCustomMetricsServerOpts(t *testing.T) {
```

Comment:
we'll probably want to refactor these tests at some point to reduce the duplication, but it's fine for now 🤓